connection.go

```diff
@@ -29,30 +29,40 @@ type StreamHandler func(stream *Stream)
 type AuthHandler func(header http.Header, slot uint8, parent uint32) bool
 
 type idleAwareFramer struct {
-	f       *spdy.Framer
-	conn    *Connection
-	expired *time.Timer
-	startCh chan struct{}
+	f              *spdy.Framer
+	conn           *Connection
+	expired        *time.Timer
+	resetChan      chan struct{}
+	setTimeoutChan chan time.Duration
+	timeout        time.Duration
 }
 
 func newIdleAwareFramer(framer *spdy.Framer) *idleAwareFramer {
 	iaf := &idleAwareFramer{
-		f:       framer,
-		startCh: make(chan struct{}, 1),
+		f:              framer,
+		resetChan:      make(chan struct{}, 2),
+		setTimeoutChan: make(chan time.Duration),
+		expired:        time.NewTimer(0),
 	}
-	go iaf.monitor()
+	iaf.expired.Stop()
```

Comment:
Good point - I could change the initial timeout 2 lines above to a value much larger than 0. My goal is to have an initialized stopped timer so I don't have to do nil checks.

Although... the unit tests pass, so maybe this is a race and I just haven't hit it yet. It's probably safer to use a value other than 0, however.